### PR TITLE
doc:  add IVSHMEM, mmio_resources, pt_intx in configuration tool

### PR DESCRIPTION
**Pull request by xiaofst**
Add configration items: IVSHMEM, mmio_resources, pt_intx in
configuration tool documentation

Signed-off-by: Shixiong Zhang <shixiongx.zhang@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: doc/tutorials/acrn_configuration_tool.rst [hunk 1]
```diff
@@ -168,6 +168,20 @@ Additional scenario XML elements:
   Specify whether force to disable software workaround for Machine Check
   Error on Page Size Change is enabled.
 
+``IVSHMEM`` (a child node of ``FEATURE``):
+  Specify the IVSHMEM configuration
+
+``IVSHMEM_ENABLED`` (a child node of ``FEATURE/IVSHMEM``):
+  Specify whether Enable Share Memory between VMs by IVSHMEM.
```
Comment (gvancuts) [inline doc/tutorials/acrn_configuration_tool.rst:175]: I feel `shared` instead of `share` would read better throughout this document. @dbkinder , thoughts?

File: doc/tutorials/acrn_configuration_tool.rst [hunk 2]
```diff
@@ -168,6 +168,20 @@ Additional scenario XML elements:
...
+
+``IVSHMEM_REGION`` (a child node of ``FEATURE/IVSHMEM``):
+  Specify the name, size, IDs of communication VMs of share memory region,
```
Comment (gvancuts) [inline doc/tutorials/acrn_configuration_tool.rst:178]:
```suggestion
  Specify the name, size, IDs of VMs communicating over the shared memory region,
```
?

File: doc/tutorials/acrn_configuration_tool.rst [hunk 3]
```diff
@@ -313,6 +327,18 @@ Additional scenario XML elements:
...
+  MMIO resources to passthrough.
+
+``TPM2`` (a child node of ``mmio_resources``):
+  TPM2 device to passthrough.
+
+``p2sb`` (a child node of ``mmio_resources``):
+  P2SB device to passthrough.
+
+``pt_intx``:
+  pt intx mapping.
```
Comment (gvancuts) [inline doc/tutorials/acrn_configuration_tool.rst:340]: This is mostly a repeat of the name of the parameter. Can you describe this a little more in plain english, i.e. Passthrough Interrupt mapping.

File: doc/tutorials/acrn_configuration_tool.rst [hunk 4]
```diff
@@ -373,6 +399,13 @@ current scenario has:
   ``bus#-port#[:bus#-port#: ...]``, e.g.: ``1-2:2-4``.
   Refer to :ref:`usb_virtualization` for details.
 
+``shm_regions``:
+  List of shared memrory regions for inter-VM communication.
```
Comment (gvancuts) [inline doc/tutorials/acrn_configuration_tool.rst:403]:
```suggestion
  List of shared memory regions for inter-VM communication.
```

File: doc/tutorials/acrn_configuration_tool.rst [hunk 5]
```diff
@@ -168,6 +168,20 @@ Additional scenario XML elements:
...
 
+``IVSHMEM`` (a child node of ``FEATURE``):
+  Specify the IVSHMEM configuration
+
+``IVSHMEM_ENABLED`` (a child node of ``FEATURE/IVSHMEM``):
+  Specify whether Enable Share Memory between VMs by IVSHMEM.
+
+``IVSHMEM_REGION`` (a child node of ``FEATURE/IVSHMEM``):
+  Specify the name, size, IDs of communication VMs of share memory region,
+  input format: hv:dm/<;shm name>;,<;shm size in MB>;,<;vm id:vm id[:vm id,...]>.
```
Comment (gvancuts) [inline doc/tutorials/acrn_configuration_tool.rst:179]: This is very hard to read... any suggestion on how we can make this more readable @dbkinder ?

File: doc/tutorials/acrn_configuration_tool.rst [hunk 6]
```diff
@@ -373,6 +399,13 @@ current scenario has:
   ``bus#-port#[:bus#-port#: ...]``, e.g.: ``1-2:2-4``.
   Refer to :ref:`usb_virtualization` for details.
 
+``shm_regions``:
+  List of shared memrory regions for inter-VM communication.
+
+``shm_region`` (a child node of ``shm_regions``):
+  configure the shm regions for current VM, input format: hv:/<;shm name>;,
```
Comment (gvancuts) [inline doc/tutorials/acrn_configuration_tool.rst:406]: Why is `dm:/` not mentionned in this section?

Comment (gvancuts) [reply doc/tutorials/acrn_configuration_tool.rst:406]: We don't need to list the VM IDs here?

**Review by gvancuts — CHANGES_REQUESTED** [review]

File: doc/tutorials/acrn_configuration_tool.rst (hunk 2 above)
Comment (dbkinder) [inline doc/tutorials/acrn_configuration_tool.rst:178]: Let's change the description to:
```
  Specify a comma-separated list of the inter-VM shared memory region name,
  size, and VM IDs that may communicate using this shared region.

  * Prefix the region ``name`` with either ``hv:/`` (for an hv-land solution) or
    ``dm:/`` for a dm-land solution; for example, ``hv:/shared01``.
    (See :ref:`ivshmem-hld` for details.)
  * Specify the region ``size`` in MB, and a power of 2 (e.g., 2, 4, 8, 16)
    up to 512.
  * Specify the two VM IDs that may use this shared memory area,
    separated by a ``:``, for example, ``0:2`` 
```
@xiaofst Your syntax description seems to allow you to share the memory area between more than two regions by writing this as, for example, ``hv:/myname,16,0:1,0:2:1:2``, but I don't think that's allowed.  Can you confirm?

File: doc/tutorials/acrn_configuration_tool.rst (hunk 3 above)
Comment (dbkinder) [reply doc/tutorials/acrn_configuration_tool.rst:340]: I agree that these descriptions could be explained better, even if it's just expanding the acronyms.

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by dbkinder** [comment]
@xiaofst Could you submit an update that addresses the review comments.  @NanlinXie

File: doc/tutorials/acrn_configuration_tool.rst (hunk 1 above)
Comment (xiaofst) [reply doc/tutorials/acrn_configuration_tool.rst:175]: ok

File: doc/tutorials/acrn_configuration_tool.rst (hunk 2 above)
Comment (xiaofst) [reply doc/tutorials/acrn_configuration_tool.rst:178]: ok

File: doc/tutorials/acrn_configuration_tool.rst (hunk 3 above)
Comment (xiaofst) [reply doc/tutorials/acrn_configuration_tool.rst:340]: I will add more information

File: doc/tutorials/acrn_configuration_tool.rst (hunk 4 above)
Comment (xiaofst) [reply doc/tutorials/acrn_configuration_tool.rst:403]: ok

File: doc/tutorials/acrn_configuration_tool.rst [hunk 7]
```diff
@@ -168,6 +168,20 @@ Additional scenario XML elements:
   Specify whether force to disable software workaround for Machine Check
   Error on Page Size Change is enabled.
 
+``IVSHMEM`` (a child node of ``FEATURE``):
```
[the remark this answers is not in the document]
Comment (xiaofst) [reply doc/tutorials/acrn_configuration_tool.rst:171]: ok

File: doc/tutorials/acrn_configuration_tool.rst (hunk 6 above)
Comment (xiaofst) [reply doc/tutorials/acrn_configuration_tool.rst:406]: > Why is `dm:/` not mentionned in this section?
the dm land solution for IVSHMEN is not officially supported in config tool, so remove dm:/ in description of IVSHMEM region and here.

> We don't need to list the VM IDs here?
Yes. It is by design of IVSHMEM.

File: doc/tutorials/acrn_configuration_tool.rst (hunk 2 above)
Comment (xiaofst) [reply doc/tutorials/acrn_configuration_tool.rst:178]: @dbkinder , synced with IVSHMEM feature owner @liuyuan-GitHub ,can we use below description:
  * Specify the VM IDs that may use this shared memory region,
     separated by ``:``, for example, ``0:2``.

The format should be hv:/<shm name>,<shm size in MB>,<vm id:vm id[:vm id:...]> . For example, hv:/myname, 16, 0:1  or hv:/myname, 16, 0:1:2

File: doc/tutorials/acrn_configuration_tool.rst (hunk 5 above)
Comment (dbkinder) [reply doc/tutorials/acrn_configuration_tool.rst:179]: my submitted edit recommendation is to use words rather than a syntax description using notation that's unfamiliar.

File: doc/tutorials/acrn_configuration_tool.rst (hunk 6 above)
Comment (dbkinder) [reply doc/tutorials/acrn_configuration_tool.rst:406]: Are you saying we only support IVSHMEM hv-land solutions via XML configuration, and only support IVSHMEM dm-land solutions via acrn-dm command line parameters?  We need to clarify this in the documentation.  @NanlinXie

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by xiaofst** [comment]
> @xiaofst Could you submit an update that addresses the review comments. @NanlinXie

yes, I have updated it.

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: doc/tutorials/acrn_configuration_tool.rst (hunk 6 above)
Comment (xiaofst) [reply doc/tutorials/acrn_configuration_tool.rst:410]: > Are you saying we only support IVSHMEM hv-land solutions via XML configuration, and only support IVSHMEM dm-land solutions via acrn-dm command line parameters? We need to clarify this in the documentation. @NanlinXie

we only support hv-land solutions, I have remove dm-land description in my update.

File: doc/tutorials/acrn_configuration_tool.rst (hunk 5 above)
Comment (xiaofst) [reply doc/tutorials/acrn_configuration_tool.rst]: > my submitted edit recommendation is to use words rather than a syntax description using notation that's unfamiliar.

yes, I have updated it as yours, the reply  is just to explain to you, it won't be added in doc.

**Review by dbkinder — APPROVED** [review]
@gvancuts to expedite this PR, I'll approve and merge it and add your edit in a separate PR.
